completers/git_completer/cmd/root.go

```diff
@@ -57,4 +62,60 @@ func init() {
 		}
 		return action
 	})
+
+	addAliasCompletion(rootCmd)
+}
+
+func addAliasCompletion(cmd *cobra.Command) {
+	if c, _, err := cmd.Find([]string{"_carapace"}); err == nil {
+		c.PreRun = func(cmd *cobra.Command, args []string) {
+			// only inject if actually completing
+			if len(args) < 2 || args[1] != rootCmd.Name() {
+				return
+			}
+
+			// pass through args related to config
+			rootCmd.ParseFlags(args[2:])
+			gitArgs := []string{}
+			if f := rootCmd.Flag("C"); f.Changed {
+				gitArgs = append(gitArgs, "-C", f.Value.String())
+			}
+			if f := rootCmd.Flag("git-dir"); f.Changed {
+				gitArgs = append(gitArgs, "--git-dir", f.Value.String())
+			}
```

Comment:
The repo itself could have aliases defined so we want to pass through any flags that potentially change the `.git` directory